src/main/java/core/basesyntax/GeometricFigureSupplier.java

```diff
@@ -0,0 +1,12 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class GeometricFigureSupplier {
+    private Random random = new Random();
+
+    public GeometricFigure getRandomFigure() {
+        int index = random.nextInt(0, GeometricFigure.values().length);
```

Comment:
The 'nextInt' method is being used incorrectly. The first parameter should be the bound (exclusive), not the starting point, which is always 0. You can simplify the call to 'nextInt(GeometricFigure.values().length)'.